smac/optimizer/smbo.py

```diff
@@ -347,3 +342,25 @@ def _get_timebound_for_intensification(self, time_spent):
                 time_spent, (1-frac_intensify), time_left, frac_intensify))
         return time_left
 
+
+class ChallengerList(object):
+
+    def __init__(self, challengers, configuration_space):
+        self.challengers = challengers
+        self.configuration_space = configuration_space
+        self._index = 0
+        self._next_is_random = False
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._index == len(self.challengers):
+            raise StopIteration
+        elif self._next_is_random:
```

Comment:
Okay, let's assume I set the parameter to draw each third configuration at random. Then, the third element in the list of challengers would be a random configuration. This would not guarantee that SMAC actually looks at a random configuration ever, because intensifiy only guarantees to look at two configurations. Therefore, I think one needs to be cautious about this.